### PR TITLE
Allow clip-box to auto expand

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
Resolves https://github.com/cogeotiff/rio-tiler/issues/607

### CODE REVIEW

**Comment by vincentsarago** [comment]
thanks @abarciauskas-bgse 🙏 

Could you add a tests for this? Thanks 😄

**Comment by abarciauskas-bgse** [comment]
Added tests: https://github.com/cogeotiff/rio-tiler/pull/608/commits/feb193658313556ad4f9c3dd2e88a91e10a22986 let me know what you think @vincentsarago

File: tests/test_io_xarray.py [hunk 1]
```diff
@@ -47,6 +50,25 @@ def test_xarray_reader():
...
+        # Tests for auto_expand
+        ## Test that a high-zoom tile will error with auto_expand=False
+        tms = morecantile.tms.get("WebMercatorQuad")
+        zoom = 10
+        x, y = tms.xy(-170, -80)
+        tile = tms._tile(x, y, zoom)
+        bounds = tms.xy_bounds(tile)
+        with pytest.raises(rioxarray.exceptions.OneDimensionalRaster) as error:
+            error = dst.tile(tile.x, tile.y, zoom)
+        assert str(error.value) == "At least one of the clipped raster x,y coordinates has only one point."
```
Comment (vincentsarago) [inline tests/test_io_xarray.py:62]:
```suggestion
            dst.tile(tile.x, tile.y, zoom)
            assert str(error.value) == "At least one of the clipped raster x,y coordinates has only one point."
```

File: tests/test_io_xarray.py [hunk 2]
```diff
@@ -47,6 +50,25 @@ def test_xarray_reader():
...
 
+        # Tests for auto_expand
+        ## Test that a high-zoom tile will error with auto_expand=False
+        tms = morecantile.tms.get("WebMercatorQuad")
+        zoom = 10
+        x, y = tms.xy(-170, -80)
+        tile = tms._tile(x, y, zoom)
+        bounds = tms.xy_bounds(tile)
+        with pytest.raises(rioxarray.exceptions.OneDimensionalRaster) as error:
+            error = dst.tile(tile.x, tile.y, zoom)
```
Comment (vincentsarago) [inline tests/test_io_xarray.py:61]:
```suggestion
            dst.tile(tile.x, tile.y, zoom)
```

**Comment by abarciauskas-bgse** [comment]
Working through the linting errors now!

**Comment by vincentsarago** [comment]
🙏 do not worry about the `benchmark` fail.

**Comment by vincentsarago** [comment]
@abarciauskas-bgse I'm still wondering if we should just set the setting to True by default! If setting it to `True` do not change the result of `non-failing` requests and avoid errors when it was previously failing I guess setting it to True will just be an enhancement 😄

**Pull request merged**